File: C++/prog_lab_4/circle.cpp
#define _CRT_SECURE_NO_WARNINGS
#include "circle.h"
#include <iostream>
#define PI 3.14159265

circle::circle():figure(1) {
}

circle::circle(float x, float y, float r) : figure(1) {//вызов в конструкторе производного класса с параметрами вызов конструктора базового класса
	pt[0].init(x, y);
	radius = r;
}

void circle::vvod()
{
	printf("Введите координаты центра окружности: ");
	pt[0].vvod();
	printf("Введите длинну радиуса окружности: ");
	while (scanf("%f", &radius) < 1) {
		while (getchar() != '\n');
		printf("Радиус введен неверно.");
	}
}

void circle::vivod()
{
	printf("Центр окружности: %.3f %.3f\nРадиус окружности: %.3f", pt[0].getx(), pt[0].gety(), radius);
}

float circle::square()
{
	return radius*radius*PI;
}

float circle::per()
{
	return 2*PI*radius;
}

File: Java/prog_lab_7/src/prog_lab_7/circle.java
package prog_lab_7;
import java.util.Scanner;

public class circle extends figure implements inputoutput , Cloneable{
	double radius;
	
	public circle() {
		super(1);
	}
	
	public circle(double x, double y, double r) {
		super(1);
		pt[0].x = x;
		pt[0].y = y;
		radius = r;
	}
	
	public void vvod() {
		this.pt[0].vvod();
		Scanner in = new Scanner(System.in);
		this.radius = in.nextDouble();
	}
	
	public void vivod()	{
		System.out.printf("(%f; %f;   %f)\n", this.pt[0].getx(), this.pt[0].gety(), this.radius);
	}

	public double square() {
            return radius*radius*Math.PI;
	}
	
	public double per() {
            return 2*Math.PI*radius;
	}

	@Override
	public circle clone() throws CloneNotSupportedException{//глубокое клонирование
		circle newcircle = (circle) super.clone();
		newcircle.pt = new point[1];
		newcircle.pt[0] = pt[0].clone();
		return newcircle;
	}
}